src/main/kotlin/study/racing/car/CarRanker.kt

```diff
@@ -0,0 +1,9 @@
+package study.racing.car
+
+class CarRanker {
+    fun getWinners(cars: List<Car>): List<Car> {
+        val sorted = cars.sortedByDescending { it.totalMoved }
+        val greatestScore = sorted.first().totalMoved
+        return sorted.filter { it.totalMoved == greatestScore }
+    }
+}
```

Comment:
CarRanker는 상태를 가지지 않으니 object로 바꾸어 주셔도 좋을 것 같아요.

그리고 CarRanker가 우승자를 판단하는 책임을 가져야 하는지도 한 번 더 생각해보시면 좋을 것 같아요.

참가한 자동차들은 우승자가 누구인지 알고 있지 않을까요?

Reply:
경주에 참가한 자동차 목록이라는 일급 컬렉션 객체가 있는 만큼 `List<Car>`보다는 정의한 일급 컬렉션을 사용해주시면 좋을 것 같아요.
위 코멘트에서도 언급했지만, CarList가 가장 멀리 간 자동차 목록을 제공한다면 다음과 같이 코드를 나타낼 수도 있을 것 같아요.
```suggestion
    fun getWinners(cars: CarList): CarList {
         return cars.getFarthestCars()
     }
```